File: bench/io/bench_readlines.php
<?php
// Count lines and bytes, one line at a time. See bench_readlines.c for the
// result fold.
$t0 = microtime(true);
$lines = 0; $total = 0;
$f = fopen("io_input.txt", "rb");
while (($line = fgets($f)) !== false) { $lines++; $total += strlen($line); }
fclose($f);
$ms = (microtime(true) - $t0) * 1000;
printf("result: %d\n", $lines * 100000000 + $total);
printf("elapsed: %d\n", (int)$ms);

File: bench/io/bench_readlines.rb
# Count lines and bytes, one line at a time. See bench_readlines.c for the
# result fold.
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
lines = 0
total = 0
File.open("io_input.txt", "rb") do |f|
  f.each_line do |line|
    lines += 1
    total += line.bytesize
  end
end
ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "result: #{lines * 100000000 + total}"
puts "elapsed: #{ms.to_i}"
